src/client/client.ts

```diff
@@ -616,6 +616,10 @@ export class Client<P = Indexable> implements Observable<ClientEvent<P>> {
       return doc;
     }
 
+    this.eventStreamObserver.next({
+      type: ClientEventType.StreamConnectionStatusChanged,
+      value: StreamConnectionStatus.Disconnected,
+    });
```

Comment:
Is there a reason to trigger StreamConnectionStatusChanged here?

`StreamConnectionStatusChanged` seems to be triggered when there is a change in the connection due to a network problem. For the same reason, `StreamConnectionStatusChanged` is not called when detaching a document.

(It seems better to shorten the name. 🤔 `StreamConnectionStatusChanged` -> `StreamStatusChanged`)

Reply:
When resuming a document, the `connected` event is triggered while running the runWatchLoop.
 https://github.com/yorkie-team/yorkie-js-sdk/blob/3b313a54eee2b20e0185f93813643479714a8244/src/client/client.ts#L913-L916
Therefore, I made sure that `disconnected` event is also triggered when pausing the document. 
`Disconnected` event is also triggered when detaching the document. 
https://github.com/yorkie-team/yorkie-js-sdk/blob/3b313a54eee2b20e0185f93813643479714a8244/src/client/client.ts#L1048-L1051

Reply:
It would be good to change the event name as part of issue #481.